### ISSUE TITLE
Opening browser in cypress open takes very long, if there are lots of other files and folders in the project

### ISSUE THREAD

**Issue by amenk** [cypress-io/cypress#26691]
### Current behavior

We use Cypress in a Magento 2 project with lots of files and folders.

```CYPRESS_BASE_URL=https://example.com/  yarn cypress open --e2e --browser chrome``` takes long time (sometimes up to 45 seconds) while in a plain project it's only 5 seconds.

Cypress seems to stat all the files, even they are not related to Cypress.

This did not seem to happen in Cypress 9 (we believe)

### Desired behavior

cypress open should always start fast for a good developer's experience.

### Test code to reproduce

1. clone https://github.com/amenk/cypress-repo-many-files
2. Execute the folder/files generator
3. Open cypress

Basically this is creating many files and folders using

```
#!/bin/bash

for i in {1..5}; do 
for j in {1..5}; do 
for k in {1..50}; do 
for l in {1..50}; do 
	FOLDER=test$i/$j/$k/$l
	mkdir -p $FOLDER
	for m in {1..5}; do 
		touch $FOLDER/$m.php
	done
done 
done 
done 
done 
```

and uses a fresh cypress project without any spec.

We also tried setting the file server folder, but this did not improve things.

### Cypress Version

12.11.0

### Node version

16.17.0

### Operating System

Ubuntu 22.04

### Other

See also: https://github.com/cypress-io/cypress/discussions/26684

**Comment by amenk** [comment]
Workaround: https://github.com/cypress-io/cypress/discussions/26684#discussioncomment-5837486

**Comment by lmiller1990** [comment]
We use `globby` to glob for things, more files leads to a linear increase in time.

It looks like we don't use the `gitIgnore` option to `globby` - I wonder if that would speed things up? We should likely use the `gitIgnore` option everywhere - I don't see why we would want to glob ignored files (like node_modules - you wouldn't have any specs there).

We could also use the `ignore` option to globby, but that would require another option in `cypress.config` - which means more documentation and more complexity. 

I'd recommend we start by adding `gitIgnore` to all globby calls. WDYT @mschile?

Would you be interested in making a PR @amenk? You'd just search the code base for `globby` and add `{ gitIgnore: true }` as an option where you find it.

**Comment by amenk** [comment]
@lmiller1990 I cant try to make a PR -- but in our project we also have tons of files which are in the git (not in `.gitignore`) so this would help only  bit. EDIT: okay, it's only 3000 files in git .. so it might actually help

Can you clarify why cypress searches in recursively the root? Is the for the fileserver ? (but I already changed the path there) ?
I would assume Cypress should only be interested in it's config files and the configured test folder? Can I also have specs outside of the cypress/ folder?

**Comment by lmiller1990** [comment]
> Can you clarify why cypress searches in recursively the root? 

I am actually not sure where the exact bottleneck is, but many of Cypress options accept a glob. I wonder if one of those is searching somewhere it shouldn't be. I'll try your reproduction and see if I can track down where the bottleneck is.

> I would assume Cypress should only be interested in it's config files and the configured test folder? Can I also have specs outside of the cypress/ folder?

I think you can put specs outside the project root - I've never actually tried this - but I'd agree that Cypress should only be globbing for the things you tell it to. Many of our defaults accept a glob, but they all recurse from the project Cypress is opened inside of.

**Comment by chrisbreiding** [comment]
@amenk, I have replicated your problem in the latest version of Cypress and will forward this ticket to the appropriate team. They will evaluate the priority of this ticket and consider their capacity to pick it up. Please note that this does not guarantee that this issue will be resolved. The ticket will indicate status changes during evaluation, so we ask that you please refrain from asking for updates. Thanks!

**Comment by FSVetaz** [comment]
I also have this issue on WSL 2 Ubuntu 20.04. It works fine on my small monorepo but crashes on my big monorepo. The workaround doesn't work for me because of some issue with parsing typescript with babel that way. 
It doesn't have the issue on mac, so my coworkers on mac do not see an issue.
